KafkaIngester/src/main/java/io/deephaven/kafka/ingest/CharFieldCopier.java

```diff
@@ -0,0 +1,26 @@
+package io.deephaven.kafka.ingest;
+
+import io.deephaven.db.v2.sources.chunk.Attributes;
+import io.deephaven.db.v2.sources.chunk.ObjectChunk;
+import io.deephaven.db.v2.sources.chunk.WritableCharChunk;
+import io.deephaven.db.v2.sources.chunk.WritableChunk;
+import io.deephaven.util.QueryConstants;
+import org.apache.avro.generic.GenericRecord;
+
+public class CharFieldCopier implements GenericRecordFieldCopier {
+    private final String fieldName;
+
+    public CharFieldCopier(String fieldName) {
+        this.fieldName = fieldName;
+    }
+
+    @Override
+    public void copyField(ObjectChunk<Object, Attributes.Values> inputChunk, WritableChunk<Attributes.Values> publisherChunk) {
+        final WritableCharChunk<Attributes.Values> output = publisherChunk.asWritableCharChunk();
+        for (int ii = 0; ii < inputChunk.size(); ++ii) {
+            final GenericRecord genericRecord =  (GenericRecord)inputChunk.get(ii);
+            final Character value = (Character)genericRecord.get(fieldName);
+            output.set(ii, value == null ? QueryConstants.NULL_CHAR : value);
```

Comment:
I was suspicious that you were doing a 0-based fill but weren't setting size. The only use I see is this, from `io.deephaven.kafka.ingest.GenericRecordChunkAdapter#handleChunk`:
```
                for (int cc = 0; cc < chunkOffsets.length; ++cc) {
                    final WritableChunk<Attributes.Values> publisherChunk = publisherChunks[chunkOffsets[cc]];
                    final int existingSize = publisherChunk.size();
                    publisherChunk.setSize(existingSize + inputChunk.size());
                    fieldCopiers[cc].copyField(inputChunk, publisherChunk);
                }
```
So the calling pattern looks like an append, but you're using `set` with a 0-based index instead of `add` or similar. Probably a bug...
If you change the field copiers, don't forget to do the non-replicated ones (String and Object).